### PR TITLE
[BE](feat): users and countries CRUD

### PR DESCRIPTION
**Pull request by anamontiaga**

### CODE REVIEW

File: api/src/modules/countries/countries.service.ts [hunk 1]
```diff
@@ -1,47 +1,45 @@
...
-      ...this.countryRepository[index],
-      name: 'Country updated',
-    };
-    return 'Country updated';
+  async update(body, id: string) {
+    const countryToUpdate = await this.countryRepository.findOneBy({
+      id: id ,
+    });
+
+    return this.countryRepository.save(countryToUpdate)
```
Comment (andresgnlez) [inline api/src/modules/countries/countries.service.ts:30]: Uhm.. I don't think you are actually updating the country here. You are just saving the same country you just found in the database. I would say you are duplicating it 😬 .

Look for an `update` method or similar in the repository and use the `body` you are receiving.

File: api/src/modules/countries/countries.service.ts [hunk 2]
```diff
@@ -1,47 +1,45 @@
...
 
-  create(country: CreateCountryDto) {
-    this.countryRepository.save(country);
+  async save(createCountryDto: CreateCountryDto) {
+    return this.countryRepository.save(createCountryDto);
   }
 
-  findAll(): Country[] {
-    return this.countryRepository.findAll();
+  async find(): Promise<Country[]> {
```
Comment (andresgnlez) [inline api/src/modules/countries/countries.service.ts:15]: `Promise<Country[]>` is not needed. It's implicit in `return this.countryRepository.find();`

File: api/src/modules/users/users.controller.ts [hunk 3]
```diff
@@ -1,29 +1,42 @@
...
+@Controller('users')
 export class UsersController {
-  @Post('users')
-  create(): string {
-    return 'This action adds a new user';
+  constructor(private usersService: UsersService) {}
+
+  @Post()
+  async save(@Body() createUserDto: CreateUserDto) {
+    this.usersService.save(createUserDto);
```
Comment (andresgnlez) [inline api/src/modules/users/users.controller.ts:20]: It's not wrong, but let's be consistent with the rest of operations and `return` the outcome of the service.

File: api/src/modules/countries/countries.service.ts [hunk 4]
```diff
@@ -1,47 +1,45 @@
...
-    if (index === -1) {
-      return 'Country not found';
-    }
-    this.countryRepository[index] = {
-      ...this.countryRepository[index],
-      name: 'Country updated',
-    };
-    return 'Country updated';
+  async update(body, id: string) {
+    const countryToUpdate = await this.countryRepository.findOneBy({
```
Comment (andresgnlez) [inline api/src/modules/countries/countries.service.ts:26]: What happens if there is no _country to update_?

File: api/src/modules/countries/countries.controller.ts [hunk 5]
```diff
@@ -1,9 +1,42 @@
...
+  }
+
+  @Get(':id')
+  async findOneBy(@Param('id') id: string) {
+    return this.countriesService.findOneBy(id);
+  }
+
+  @Put(':id')
+  async update(@Param('id') id: string) {
+    return this.countriesService.update(id);
```
Comment (andresgnlez) [inline api/src/modules/countries/countries.controller.ts:35]: A `body` is probably missing here.

File: api/src/modules/users/users.controller.ts [hunk 6]
```diff
@@ -1,9 +1,42 @@
...
+  }
+
+  @Get()
+  async find(): Promise<User[]> {
```
Comment (andresgnlez) [inline api/src/modules/users/users.controller.ts:24]: https://github.com/Vizzuality/4-growth/pull/13/files#r1619085079

File: api/src/modules/users/users.controller.ts (hunk 3 above)
Comment (alexeh) [reply api/src/modules/users/users.controller.ts:20]: It is actually wrong, as you are neither awaiting for the process to kick back from the even loop, neither returning any value for the same purpose. 

You could potentially run into race conditions like this

**Pull request merged**